sentry-quartz/src/main/java/io/sentry/quartz/SentryJobListener.java

```diff
@@ -0,0 +1,207 @@
+import io.sentry.SentryIntegrationPackageStorage;
+import io.sentry.SentryLevel;
+import io.sentry.protocol.SentryId;
+import java.util.List;
+import java.util.TimeZone;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.quartz.CalendarIntervalTrigger;
+import org.quartz.CronTrigger;
+import org.quartz.DateBuilder;
+import org.quartz.Job;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.quartz.JobListener;
+import org.quartz.SimpleTrigger;
+import org.quartz.Trigger;
+
+public final class SentryJobListener implements JobListener {
+
+  public static final String SENTRY_CHECK_IN_ID_KEY = "sentry-checkin-id";
+  public static final String SENTRY_CHECK_IN_SLUG_KEY = "sentry-checkin-slug";
+
+  public SentryJobListener() {
+    SentryIntegrationPackageStorage.getInstance().addIntegration("Quartz");
+    SentryIntegrationPackageStorage.getInstance()
+        .addPackage("maven:io.sentry:sentry-quartz", BuildConfig.VERSION_NAME);
+  }
+
+  @Override
+  public String getName() {
+    return "sentry-job-listener";
+  }
+
+  @Override
+  public void jobToBeExecuted(JobExecutionContext context) {
+    try {
+      final @NotNull String slug = getSlug(context.getJobDetail());
+      final @NotNull CheckIn checkIn = new CheckIn(slug, CheckInStatus.IN_PROGRESS);
+
+      final @Nullable MonitorConfig monitorConfig = extractMonitorConfig(context);
+      if (monitorConfig != null) {
+        checkIn.setMonitorConfig(monitorConfig);
+      }
+
+      final @NotNull SentryId checkInId = Sentry.captureCheckIn(checkIn);
+      context.put(SENTRY_CHECK_IN_ID_KEY, checkInId);
+      context.put(SENTRY_CHECK_IN_SLUG_KEY, slug);
+    } catch (Throwable t) {
+      Sentry.getCurrentHub()
+          .getOptions()
+          .getLogger()
+          .log(SentryLevel.ERROR, "Unable to capture check-in in jobToBeExecuted.", t);
+    }
+  }
+
+  private @NotNull String getSlug(final @Nullable JobDetail jobDetail) {
+    if (jobDetail == null) {
+      return "fallback";
+    }
+    final @NotNull StringBuilder slugBuilder = new StringBuilder();
+
+    final @Nullable JobKey key = jobDetail.getKey();
+    if (key != null) {
+      slugBuilder.append(key.getName());
+    }
+
+    slugBuilder.append("__");
+
+    final @Nullable Class<? extends Job> jobClass = jobDetail.getJobClass();
+    if (jobClass != null) {
+      slugBuilder.append(jobClass.getCanonicalName());
+    }
+
+    return slugBuilder.toString();
+  }
+
+  private @Nullable MonitorConfig extractMonitorConfig(final @NotNull JobExecutionContext context) {
+    @Nullable MonitorSchedule schedule = null;
+    @Nullable String cronExpression = null;
+    @Nullable TimeZone timeZone = TimeZone.getDefault();
+    @Nullable Integer repeatInterval = null;
+    @Nullable MonitorScheduleUnit timeUnit = null;
+
+    try {
+      List<? extends Trigger> triggersOfJob =
```

Comment:
Shouldn't we use `context.getTrigger()` instead?
Maybe we'd also have to add some part of the trigger to the monitor slug.
Because the we way it is now, if we have two or more triggers for the job it will always take the info of the last `CronTrigger` in the `triggersOfJob` list.

Small example (add to SentryDemoApplication in jakarta):

```
  @Bean
  public JobDetailFactoryBean jobDetail() {
    JobDetailFactoryBean jobDetailFactory = new JobDetailFactoryBean();
    jobDetailFactory.setName("hello there 123");
    jobDetailFactory.setJobClass(SampleJob.class);
    jobDetailFactory.setDescription("Invoke Sample Job service...");
    jobDetailFactory.setDurability(true);
    return jobDetailFactory;
  }

  @Bean
  public SimpleTriggerFactoryBean trigger(JobDetail job) {
    SimpleTriggerFactoryBean trigger = new SimpleTriggerFactoryBean();
    trigger.setJobDetail(job);
    trigger.setRepeatInterval(10000);
    trigger.setRepeatCount(SimpleTrigger.REPEAT_INDEFINITELY);
    return trigger;
  }

    @Bean
    public CronTriggerFactoryBean trigger2(JobDetail job) {
      CronTriggerFactoryBean trigger = new CronTriggerFactoryBean();
      trigger.setJobDetail(job);
      trigger.setCronExpression("0/20 * * ? * *");
      return trigger;
    }
```

Reply:
At the moment there's no support for multiple triggers in Sentry. I opted for a more consistent approach by getting the list of triggers so we don't update the monitor config on every invocation.